src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ApplicableLanguageAttribute.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.DotNet.UpgradeAssistant
+{
+    using System;
+
+    /// <summary>
+    /// An attribute for marking code fix providers that are only applicable when
+    /// the project being upgraded contains certain language.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = false)]
```

Comment:
It may make sense to allow multiple in case something in the future works on, say, C# and F# but not VB or something weird like that.

Reply:
At the moment, if something applies to multiple languages then it wouldn't use the ApplicableLanguage filter. The default is opt-in to all languages.

Decided that if we need to filter something that applies to VB and C# but not F# that would be a design decision, and unit tests, for later. LMK if you think it's worth considering as part of this change.